boa/src/builtins/reflect/mod.rs

```diff
@@ -0,0 +1,373 @@
+        result.set_field("length", 0, context)?;
+
+        let keys = target.own_property_keys();
+        for (i, k) in keys.iter().enumerate() {
+            result.set_field(i, k, context)?;
+        }
+
+        Ok(result)
+    }
+
+    /// Prevents new properties from ever being added to an object.
+    ///
+    /// More information:
+    ///  - [ECMAScript reference][spec]
+    ///  - [MDN documentation][mdn]
+    ///
+    /// [spec]: https://tc39.es/ecma262/#sec-reflect.preventextensions
+    /// [mdn]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Reflect/preventExtensions
+    pub(crate) fn prevent_extensions(
+        _: &Value,
+        args: &[Value],
+        context: &mut Context,
+    ) -> Result<Value> {
+        let mut target = args
+            .get(0)
+            .and_then(|v| v.as_object())
+            .ok_or_else(|| context.construct_type_error("target must be an object"))?;
+
+        Ok(target.prevent_extensions().into())
+    }
+
+    /// Sets a property of an object.
+    ///
+    /// More information:
+    ///  - [ECMAScript reference][spec]
+    ///  - [MDN documentation][mdn]
+    ///
+    /// [spec]: https://tc39.es/ecma262/#sec-reflect.set
+    /// [mdn]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Reflect/set
+    pub(crate) fn set(_: &Value, args: &[Value], context: &mut Context) -> Result<Value> {
+        let undefined = Value::undefined();
+        let mut target = args
+            .get(0)
+            .and_then(|v| v.as_object())
+            .ok_or_else(|| context.construct_type_error("target must be an object"))?;
+        let key = args.get(1).unwrap_or(&undefined).to_property_key(context)?;
+        let value = args.get(2).unwrap_or(&undefined);
+        if let Some(receiver) = args.get(3) {
+            target = receiver
+                .as_object()
+                .ok_or_else(|| context.construct_type_error("receiver must be an object"))?;
+        }
+        Ok(target.set(key, value.clone(), context)?.into())
+    }
+
+    /// Sets the prototype of an object.
+    ///
+    /// More information:
+    ///  - [ECMAScript reference][spec]
+    ///  - [MDN documentation][mdn]
+    ///
+    /// [spec]: https://tc39.es/ecma262/#sec-reflect.setprototypeof
+    /// [mdn]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Reflect/setPrototypeOf
+    pub(crate) fn set_prototype_of(
+        _: &Value,
+        args: &[Value],
+        context: &mut Context,
+    ) -> Result<Value> {
+        let undefined = Value::undefined();
+        let mut target = args
+            .get(0)
+            .and_then(|v| v.as_object())
+            .ok_or_else(|| context.construct_type_error("target must be an object"))?;
+        let proto = args.get(1).unwrap_or(&undefined);
+        if !proto.is_null() && !proto.is_object() {
+            return Err(context.construct_type_error("proto must be an object or null"));
```

Comment:
Might aswell use `throw_type_error`, less text. There are 2 other instances in the file.
```suggestion
            return context.throw_type_error("proto must be an object or null");
```